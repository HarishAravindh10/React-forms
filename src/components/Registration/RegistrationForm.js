import React, { useState, useRef } from 'react';

function RegistrationForm() {
  const [id, setID] = useState('');
  const [name, setName] = useState('');
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');
  const [file, setFile] = useState(null);
  const formRef = useRef(null);

  const handleIDChange = (e) => {
    const value = e.target.value;
    if (/^\d*$/.test(value)) {
      setID(value);
    }
  };

  const handleNameChange = (e) => {
    const value = e.target.value;
    if (/^[a-zA-Z]*$/.test(value)) {
      setName(value);
    }
  };

  const handleUsernameChange = (e) => {
    setUsername(e.target.value);
  };

  const handlePasswordChange = (e) => {
    setPassword(e.target.value);
  };

  const handleFileChange = (e) => {
    setFile(e.target.files[0]);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    console.log('ID:', id);
    console.log('Name:', name);
    console.log('Username:', username);
    console.log('Password:', password);
    console.log('File:', file);

    alert('Form is submitted');

    // Clear form fields after submission
    formRef.current.reset();
    setID('');
    setName('');
    setUsername('');
    setPassword('');
    setFile(null);
  };

  return (
    <div className='RegistrationForm'>
      <h2> Registration Form </h2>
      <form ref={formRef} onSubmit={handleSubmit}>
        <div>
          <label htmlFor='id'>ID:</label>
          <input
            id="id"
            type="text"
            value={id}
            onChange={handleIDChange}
            pattern="\d*"
          />
        </div>
        <div>
          <label>Name:</label>
          <input
            type="text"
            value={name}
            onChange={handleNameChange}
          />
        </div>
        <div>
          <label>Username:</label>
          <input
            type="text"
            value={username}
            onChange={handleUsernameChange}
          />
        </div>
        <div>
          <label>Password:</label>
          <input
            type="password"
            value={password}
            onChange={handlePasswordChange}
          /> 
        </div>
        <div>
          <label>Upload File:</label>  
          <input
            type="file"
            onChange={handleFileChange}
          />
        </div>
        <button type="submit">Submit</button>
      </form>
    </div>
  );
}

export default RegistrationForm;
