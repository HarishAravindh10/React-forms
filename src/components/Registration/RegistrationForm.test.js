// import React from 'react';
// import { render, fireEvent, screen } from '@testing-library/react';
// import RegistrationForm from './RegistrationForm';
import React from 'react';
import renderer from 'react-test-renderer';
import RegistrationForm from './RegistrationForm';

test('renders registration form', () => {
  const component = renderer.create(<RegistrationForm />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


// test('updates ID input field correctly', () => {
//   render(<RegistrationForm />);
//   const idInput = screen.getByLabelText('ID:');
//   fireEvent.change(idInput, { target: { value: '123' } });
//   expect(idInput.value).toBe('123');
// });

// test('handles invalid characters in ID input field', () => {
//   render(<RegistrationForm />);
//   const idInput = screen.getByLabelText('ID:');
//   fireEvent.change(idInput, { target: { value: 'abc' } });
//   expect(idInput.value).toBe('');
// });

// // Write more test cases for other input fields and form submission
