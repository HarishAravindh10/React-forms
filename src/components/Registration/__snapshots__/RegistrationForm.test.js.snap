// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders registration form 1`] = `
<div
  className="RegistrationForm"
>
  <h2>
     Registration Form 
  </h2>
  <form
    onSubmit={[Function]}
  >
    <div>
      <label
        htmlFor="id"
      >
        ID:
      </label>
      <input
        id="id"
        onChange={[Function]}
        pattern="\\d*"
        type="text"
        value=""
      />
    </div>
    <div>
      <label>
        Name:
      </label>
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <label>
        Username:
      </label>
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <label>
        Password:
      </label>
      <input
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div>
      <label>
        Upload File:
      </label>
      <input
        onChange={[Function]}
        type="file"
      />
    </div>
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
